app.rb

```diff
@@ -43,6 +43,19 @@
 end
 
 #
+#  Fetches a random definition by following the /random.php redirect
+#  and taking out the term with a regexp
+#
+get '/random' do
+  rdef = ud.get_random_definition()
+    unless valid_response? definition
+    raise NoDefinition, params[:term]
+  end
+
+  jsonp definition
```

Comment:
So here we go: https://github.com/nickcharlton/urbanscraper/pull/7

I reworked the tests to use [RSpec](https://rspec.info), as that's what I mostly write these days and also simplified them a bit. I hope these will help you get going!